server/enterprise/elasticsearch/elasticsearch/bulk.go

```diff
@@ -69,12 +79,20 @@ func (r *Bulk) DeleteOp(op types.DeleteOperation) error {
 		return err
 	}
 
-	return r.flushIfNecessary()
+	if !r.sync {
+		return r.flushIfNecessary()
+	}
+	return nil
 }
 
 // flushIfNecessary flushes the pending buffer if needed.
 // It MUST be called with an already acquired mutex.
 func (r *Bulk) flushIfNecessary() error {
+	// Flush always if in sync mode
+	if r.sync {
```

Comment:
Do we still need this flag? It is already checked at the higher level.

Reply:
We can remove it, it was some defensive code but it's true that it's not necessary.

Reply:
Fixed in https://github.com/mattermost/mattermost/pull/33400/commits/126e60c6e32c95c94f78986a8544576d3e537445.